### PR TITLE
Add Bounding Box Utils

### PR DESCRIPTION
**Pull request by sineeli**
Added below utils for bounding boxes:

- [x] `converters`
- [x] `to_dense`
- [x] `to_ragged`

Future utils needed to be added:

- [ ]  `iou`
- [ ] `mask_invalid_detections`
- [ ]  `ensure_tensor`
- [ ]  extra `utils`

### CODE REVIEW

**Comment by mattdangerw** [comment]
Please run format!

@divyashreepathihalli this is the temporary move to keras-hub right?

File: keras_nlp/src/bounding_box/converters_test.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+import numpy as np
+import pytest
+from absl.testing import parameterized
+from keras import backend
+
+from keras_nlp.src import bounding_box
+from keras_nlp.src.tests.test_case import TestCase
+
+try:
```
Comment (mattdangerw) [inline keras_nlp/src/bounding_box/converters_test.py:25]: no need to conditional tf import in tests

File: keras_nlp/src/bounding_box/converters_test.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    import tensorflow as tf
+except ImportError:
+    tf = None
+
+xyxy_box = np.array([[[10, 20, 110, 120], [20, 30, 120, 130]]], dtype="float32")
```
Comment (mattdangerw) [inline keras_nlp/src/bounding_box/converters_test.py:30]: i'd move all these to inside the test case setUp or individual tests, we don't want import to create a bunch of arrays on file import.

**Review by mattdangerw — APPROVED** [review]
My understanding is this is just a temporary home to get ourselves unblocked. If so, lgtm!

**Comment by sineeli** [comment]
> Please run format!


I did run the format but seems the error is coming from api_gen. I even ran api_gen.sh but not sure what is causing the error.

Thanks for the review, I will resolve the issues mentioned.

Thanks!

**Comment by divyashreepathihalli** [comment]
looks like it needs `shell/api_gen.sh`, maybe try it again?

**Review by divyashreepathihalli — APPROVED** [review]

**Comment by divyashreepathihalli** [comment]
> Please run format!
> 
> @divyashreepathihalli this is the temporary move to keras-hub right?

Yes! this should be temporary until they are added to keras core

File: keras_nlp/src/bounding_box/converters_test.py (hunk 1 above)
Comment (sineeli) [reply keras_nlp/src/bounding_box/converters_test.py]: Done

File: keras_nlp/src/bounding_box/converters_test.py (hunk 2 above)
Comment (sineeli) [reply keras_nlp/src/bounding_box/converters_test.py]: Done

**Pull request merged**